torch/distributed/_tensor/dispatch.py

```diff
@@ -129,21 +148,32 @@ def operator_dispatch(
     suggested_input_schema = output_sharding.schema_suggestions[0]
     needs_redistribute = suggested_input_schema is not op_schema
 
-    local_tensor_args = pack_args_kwargs_with_local_tensor(
-        args,
-        suggested_input_schema.args_schema,
-        redistribute_with_schema=needs_redistribute,
-    )
-    local_tensor_kwargs = pack_args_kwargs_with_local_tensor(
-        kwargs,
-        suggested_input_schema.kwargs_schema,
-        redistribute_with_schema=needs_redistribute,
-    )
-
-    # run local op computation with potentially modified args/kwargs
-    local_tensor_args = cast(Tuple[object, ...], local_tensor_args)
-    local_tensor_kwargs = cast(Dict[str, object], local_tensor_kwargs)
-    local_results = op_call(*local_tensor_args, **local_tensor_kwargs)
+    if mesh is not None and mesh.get_coordinate() is None:
+        # if we are on a non-participating device, we simply return
+        # an empty tensor for now.
+        # TODO: what if the op returns a non-tensor value, what if
+        # the op returns a list of tensors, we need to figure out
+        # a consistent way to handle that, and also need to figure
+        # out if we should communicate the result to non-participating
+        # ranks (i.e. a.sum() -> scalar, maybe we should set to 0)
+        local_results = torch.tensor([])
```

Comment:
1. Communicating the result to non-participating ranks will block those ranks so we may not want to do it unless necessary. If we really want some result on those ranks, we can use a default value passed by users but that requires a change of all DTensor ops argument list by adding a default value.

2. The op return type problem should also be fixed by using a default value passed in.

Reply:
I think we should set the default values by ourselves, I will submit a follow up PR later to address this.